### PR TITLE
Initial Release of New "Card" Component (aka Card Replacement)

### PR DESCRIPTION
**Pull request by sghoweri**
## Jira
http://vjira2:8080/browse/BDS-1944
http://vjira2:8080/browse/BDS-685

## Summary
Ports over and cleans up the major Card refactor work from https://github.com/boltdesignsystem/bolt/pull/928.

## Details
The vast majority of the work here is largely the same from the original PR which includes things like:

* Simpler, modernized API that moves us away from the `pattern` approach that's been around since the initial Card was released
* Refactored Twig templates and CSS to accommodate and allow for free-form content / much more flexible ways to use Cards
* Full web component support (including SSR support via Twig)
* Whole card clickability

Beyond this, a handful of additional updates are included in this PR, in addition to a few others that we'll continue to iterate on beyond this initial release.

### Completed Updates

1. Continued legacy support of the Old (original) Card component in Bolt v2.x -- aka the new Card and the old Legacy Card can both continue to be used in the exact same codebase without requiring **any** API updates to original Card instances.

This 100% guaranteed backwards compatibility has been accomplished by temporarily renaming all CSS classes, Twig templates and custom elements used in this new Card component replacement as "Card Replacement" until we hit Bolt v3.0. 

While this update  unfortunate however READMEs, package.json descriptions and Pattern Lab docs / Nav have all been updated to accommodate  this as best we can.

2. The vast majority of the old Card-specific demos have been removed + the docs have been updated to point users looking for docs on the old Card component back to `v2.13.0` to reduce the amount of tech debt being taken on + communicate that the old Card component is no longer being updated with any new features.

3. The original Card adapter (painstakingly created to allow for a more painless transition over to the new Card) has been removed. 

While the old Card component code still technically remains in the Bolt codebase (for required hotfixes), we are no longer going to be supporting an automatic transition over to the new Card component due to the required name change + early release in Bolt v2.x without requiring any tech debt cleanup in the process.

As communicated since late 2018 (yes, seriously, it's been that long) Bolt v3.0 will 100% remove all support for the old Card API and will require consumers to migrate over to this new canonical Card component, including instances of hard-coded components that are not covered by Twig or Web Component templating.

4. The Pattern Lab Nav for this new Card component has been slightly updated for clearer names:
image

5. Minor API / dependency / syntax changes have been adjusted to better reflect some of our more recent components.
 

### Preliminary Updates

6. Preliminary support for `horizontal` layouts (originally being spec'd out and iterated on for the in-flight Academy work, requested by several people)
 
image

7. In-progress migration to LitElement

While this work is still ongoing, some of the components / subcomponents used have been upgraded to use our new LitElement-based base class that brand new components use out of the gate.

#### Already Migrated
- `<bolt-card-replacement>`
- `<bolt-card-replacement-body>`

#### Pending Migration
- `<bolt-card-replacement-media>`
- `<bolt-card-replacement-link>`
- `<bolt-card-replacement-actions>`
- `<bolt-card-replacement-action>`

8. Slot Support

This work is still getting iterated on however in practice this will provide an even simpler way to use the new Card web component in the most common of use cases.

Instead of requiring this:
```
<bolt-card>
  <bolt-card-media>
    <img src="cat.jpg">
  </bolt-card-media>
  <bolt-card-body>
    <p>Example body text goes here</p>
  </bolt-card-body>
  <bolt-card-actions>
    <bolt-card-action url="https://pega.com">Internal link</bolt-card-action>
  </bolt-card-actions>
</bolt-card>
```

You'll instead be able to do this:
```
<bolt-card>
  <img src="cat.jpg slot="media">
  <p slot="body">Example body text goes here</p>
  <bolt-button url="https://pega.com" color="text" slot="action">Internal link</bolt-button>
</bolt-card>
```

## How to test
- At a high level quickly re-review the changes _however_ keep in mind that the bulk of the work here has already been reviewed and approved via previous work done.

The goal is for this to be an early release to support Academy that we can continue to iterate on as we work up to Bolt v3.0!

### CODE REVIEW

File: docs-site/src/pages/pattern-lab/_patterns/02-components/text/20-text-display-variations.twig [hunk 1]
```diff
@@ -1,20 +0,0 @@
-{% extends "@pl/_text-doc-template.twig" %}
```
Comment (mikemai2awesome) [inline docs-site/src/pages/pattern-lab/_patterns/02-components/text/20-text-display-variations.twig]: Why is this page removed?

Comment (sghoweri) [reply docs-site/src/pages/pattern-lab/_patterns/02-components/text/20-text-display-variations.twig]: 🤷‍♂super weird... not sure. 

I'll get this file restored!

Comment (sghoweri) [reply docs-site/src/pages/pattern-lab/_patterns/02-components/text/20-text-display-variations.twig]: Should be all set @mikemai2awesome

**Comment by sghoweri** [comment]
> @sghoweri your fix worked for the IE11 layout issue on the Theme Demo 👍
> 
> Some other notes...
> 
> **Possible regression?**
> In IE11, check out the first card in the top row: [deprecated card demo](https://feature-rename-card-refactor.boltdesignsystem.com/pattern-lab/patterns/02-components-card-deprecated--01-deprecated-card/02-components-card-deprecated--01-deprecated-card.html). As the component renders you see the button padding disappear and reappear. It looks like it is styled initially, then becomes unstyled, then it is restyled. There is not this type of FOUC on the [latest release](https://boltdesignsystem.com/pattern-lab/patterns/02-components-card-25-card-band-variation/02-components-card-25-card-band-variation.html). Can you reproduce it? I've seen this type of thing flagged by QA before. Any idea what might've changed?
> 
> **Important to fix (but not blocker for merge)**
> In IE11, the SSR (or interim) version is significantly different than final rendered version. Each card layout jumps around a lot as it loads. Many of them seem to go through two cycles of rendering, each causing FOUC and a lot of page jumpiness. I think this will be important to address before these cards are released on Academy. Thoughts?

@danielamorse none of the code for the original Card changed with this PR, remember?

This looks like that SSR behavior it's happening on master atm so I'd argue that that shouldn't at least hold up getting this down to master. I'm checking to see if my fix for http://vjira2:8080/browse/BDS-1946 might help with this!

**Comment by sghoweri** [comment]
PR was released with v2.14.0

**Pull request merged**